src/core/plugins/core/plugin.spec.js

```diff
@@ -209,11 +209,59 @@ describe("core plugin", () => {
   });
 
   describe("action__unpack()", () => {
-    it("should unpack", () =>
-      core.action__unpack({
-        group: "firmware",
-        files: [{ archive: "a.zip", dir: "a" }]
-      })); // TODO add assertions
+    // TODO add assertions
```

Comment:
I don't think we should check exact params to the emits - imo we should test the functionality, not the implementation and I'd say the tests should not fail if we change the animation or wording on the messages - checking whether emit was called 3 times is enough I think?

Reply:
i followed the other tests. to actually test if it unpacks we would need to include a test file.
and if we just mock it all away, it kind of is useless as well?

not too common with tests, so will go the way you think is best.

Reply:
You should mock this `unpack`:
```
const { unpack } = require("../../helpers/asarLibs.js");
```
and then check if it was called with correct arguments

Reply:
Removed them and changed the todo message.

Reply:
is it fine like this now?